src/lib/block_producer/block_producer.ml

```diff
@@ -1239,7 +1248,11 @@ let run ~context:(module Context : CONTEXT) ~vrf_evaluator ~prover ~verifier
             ]
           "Node started before genesis: waiting $time_till_genesis \
            milliseconds before starting block producer" ;
-      upon (schedule ~time_controller genesis_state_timestamp) start )
```

Comment:
These two code pieces are 100% equivalent, no?

And arguably `upon` is a bit more straightforward.